### PR TITLE
Android: handle location service is off case and return error

### PR DESCRIPTION
**Pull request by khalidaw**
### Platforms affected
Android

### What does this PR do?
Return error when location is off, same error as return in iOS

### What testing has been done on this change?
Bashed application using this version with this change

### Checklist
- [ ] [Reported an issue](http://cordova.apache.org/contribute/issues.html) in the JIRA database
- [ ] Commit message follows the format: "CB-3232: (android) Fix bug with resolving file paths", where CB-xxxx is the JIRA ID & "android" is the platform affected.
- [ ] Added automated test coverage as appropriate for this change.

### CODE REVIEW

**Comment by cordova-qa** [comment]
Cordova CI Build has completed successfully.

**Commit**     - [Link](https://github.com/apache/cordova-plugin-geolocation/pull/85/commits/582abd533a0e3a3bf1f70ea34bee221f8a25e2ed)
**Dashboard** - [Link](http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/28/)

| Builder Name  | Console Output | Test Report | Device Logs  |
|     :---:     |     :---:      |   :---:     |     :---:    |
| [Windows 8.1 Store]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/28//PLATFORM=windows-8.1-store/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/28//PLATFORM=windows-8.1-store/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/28//PLATFORM=windows-8.1-store/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/28//PLATFORM=windows-8.1-store/artifact/) |
| [Windows 10  Store]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/28//PLATFORM=windows-10-store/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/28//PLATFORM=windows-10-store/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/28//PLATFORM=windows-10-store/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/28//PLATFORM=windows-10-store/artifact/) |
| [Windows 8.1 Phone]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/28//PLATFORM=windows-8.1-phone/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/28//PLATFORM=windows-8.1-phone/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/28//PLATFORM=windows-8.1-phone/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/28//PLATFORM=windows-8.1-phone/artifact/) |
| [iOS]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/28//PLATFORM=ios/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/28//PLATFORM=ios/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/28//PLATFORM=ios/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/28//PLATFORM=ios/artifact/) |
| [Android]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/28//PLATFORM=android/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/28//PLATFORM=android/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/28//PLATFORM=android/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/28//PLATFORM=android/artifact/) |

**Comment by cordova-qa** [comment]
Cordova CI Build has completed successfully.

**Commit**     - [Link](https://github.com/apache/cordova-plugin-geolocation/pull/85/commits/94174c6edbb074cce11e52910d1e6a0d5b0b50b0)
**Dashboard** - [Link](http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/29/)

| Builder Name  | Console Output | Test Report | Device Logs  |
|     :---:     |     :---:      |   :---:     |     :---:    |
| [Windows 8.1 Store]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/29//PLATFORM=windows-8.1-store/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/29//PLATFORM=windows-8.1-store/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/29//PLATFORM=windows-8.1-store/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/29//PLATFORM=windows-8.1-store/artifact/) |
| [Windows 10  Store]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/29//PLATFORM=windows-10-store/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/29//PLATFORM=windows-10-store/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/29//PLATFORM=windows-10-store/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/29//PLATFORM=windows-10-store/artifact/) |
| [Windows 8.1 Phone]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/29//PLATFORM=windows-8.1-phone/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/29//PLATFORM=windows-8.1-phone/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/29//PLATFORM=windows-8.1-phone/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/29//PLATFORM=windows-8.1-phone/artifact/) |
| [iOS]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/29//PLATFORM=ios/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/29//PLATFORM=ios/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/29//PLATFORM=ios/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/29//PLATFORM=ios/artifact/) |
| [Android]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/29//PLATFORM=android/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/29//PLATFORM=android/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/29//PLATFORM=android/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-geolocation-pr/29//PLATFORM=android/artifact/) |

**Comment by stevengill** [comment]
Hey @khalidaw!

Is there an issue for this? issues.cordova.io

**Comment by lincolnthree** [comment]
If creating a JIRA issue for this PR is the only thing holding it up, here you go...

https://issues.apache.org/jira/browse/CB-12406

**Comment by Menardi** [comment]
@stevengill Is there anything else that needs to be done to get this merged? It seems like quite a key feature to me!

**Comment by RickeyWard** [comment]
This branch needs merged, it fixes inconsistencies between android and ios and after hours of trying to get android to work with no avail I installed this fork and it worked immediately. yes we can all download it from his fork, but it should be in the main repo. It's a definite improvement.

**Comment by infil00p** [comment]
Why are you returning no error message when you're sending back the ILLEGAL_ACCESS error when the permission is denied?

**Comment by gianpaj** [comment]
ping to @khalidaw . See comment from infil00p

**Review by willqz — APPROVED** [review]

**Review by willqz — APPROVED** [review]

File: src/android/Geolocation.java [hunk 1]
```diff
@@ -102,6 +113,16 @@ public void requestPermissions(int requestCode)
         PermissionHelper.requestPermissions(this, requestCode, permissions);
     }
 
-
+	private boolean isGPSdisabled() {
+		boolean gps_enabled;
+		try {
+			gps_enabled = mLocationManager.isProviderEnabled(LocationManager.GPS_PROVIDER);
```
Comment (breautek) [inline src/android/Geolocation.java:119]: This PR will effectively enforce the GPS hardware requirement by requiring GPS_PROVIDER to be enabled.

The GPS_PROVIDER is what provides the GPS updates via satellites, but this isn't the only way to obtain GPS updates (however it is the most accurate once the triangulation is fixed). Forcing GPS_PROVIDER however is an application detail, not something that should be done at a framework/library level.

`GPS_PROVIDER` isn't an indication that location services is "off".

The proper way to check if location services is enabled is via: https://developer.android.com/reference/androidx/core/location/LocationManagerCompat#isLocationEnabled(android.location.LocationManager)

**Pull request closed without merging**